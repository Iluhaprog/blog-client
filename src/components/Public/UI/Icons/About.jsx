import React from 'react';
import {
  ReactComponent as AboutIcon,
} from '../../../../assets/public/icons/about.svg';
import PropTypes from 'prop-types';
import {Icon} from './Icon';

const About = ({width, height}) => (
  <Icon
    width={width}
    height={height}
  >
    <AboutIcon />
  </Icon>
);

About.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
};

export {About};
