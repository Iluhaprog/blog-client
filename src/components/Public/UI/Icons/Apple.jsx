import React from 'react';
import {
  ReactComponent as AppleIcon,
} from '../../../../assets/public/icons/apple.svg';
import PropTypes from 'prop-types';
import {Icon} from './Icon';

const Apple = ({width, height}) => (
  <Icon
    width={width}
    height={height}
  >
    <AppleIcon />
  </Icon>
);

Apple.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
};

export {Apple};
