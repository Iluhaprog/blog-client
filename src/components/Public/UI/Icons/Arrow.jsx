import React from 'react';
import {
  ReactComponent as ArrowIcon,
} from '../../../../assets/public/icons/scroll-arrow.svg';
import PropTypes from 'prop-types';
import {Icon} from './Icon';

const Arrow = ({width, height}) => (
  <Icon
    width={width}
    height={height}
  >
    <ArrowIcon />
  </Icon>
);

Arrow.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
};

export {Arrow};
