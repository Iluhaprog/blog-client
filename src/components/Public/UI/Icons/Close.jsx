import React from 'react';
import {
  ReactComponent as CloseIcon,
} from '../../../../assets/public/icons/close.svg';
import PropTypes from 'prop-types';
import {Icon} from './Icon';

const Close = ({width, height}) => (
  <Icon
    width={width}
    height={height}
  >
    <CloseIcon />
  </Icon>
);

Close.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
};

export {Close};
