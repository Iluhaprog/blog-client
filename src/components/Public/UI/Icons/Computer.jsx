import React from 'react';
import {
  ReactComponent as ComputerIcon,
} from '../../../../assets/public/icons/computer.svg';
import PropTypes from 'prop-types';
import {Icon} from './Icon';

const Computer = ({width, height}) => (
  <Icon
    width={width}
    height={height}
  >
    <ComputerIcon />
  </Icon>
);

Computer.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
};

export {Computer};
