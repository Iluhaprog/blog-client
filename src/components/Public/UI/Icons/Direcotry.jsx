import React from 'react';
import {
  ReactComponent as DirectoryIcon,
} from '../../../../assets/public/icons/directory.svg';
import PropTypes from 'prop-types';
import {Icon} from './Icon';

const Directory = ({width, height}) => (
  <Icon
    width={width}
    height={height}
  >
    <DirectoryIcon />
  </Icon>
);

Directory.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
};

export {Directory};
