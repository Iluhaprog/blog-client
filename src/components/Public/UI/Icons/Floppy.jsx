import React from 'react';
import {
  ReactComponent as FloppyIcon,
} from '../../../../assets/public/icons/floppy.svg';
import PropTypes from 'prop-types';
import {Icon} from './Icon';

const Floppy = ({width, height}) => (
  <Icon
    width={width}
    height={height}
  >
    <FloppyIcon />
  </Icon>
);

Floppy.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
};

export {Floppy};
