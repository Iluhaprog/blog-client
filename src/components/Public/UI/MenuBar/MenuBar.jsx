import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

const MenuBarBox = styled.div`
  position: fixed;
  display: flex;
  width: 100%;
  min-height: 5vh;
  background-color: #fff;
  box-shadow: 0 1px 0 #000000;
`;

const MenuBarLogo = styled.div`
  svg {
    fill: #000;
  }
  &:hover {
    background-color: #000;
    svg {
      color: #fff;
      fill: #fff;
    }
  }
`;

const MenuBarNav = styled.nav`
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 5vh;
  font-family: var(--public-font);
  color: #000;

  a {
    margin: 0 5px;
    color: #000;
    padding: 5px;

    &:hover {
      background-color: #000;
      color: #fff;
      text-decoration: none;
    }
  }
`;

const MenuBar = ({children, logo: Logo}) => {
  return (
    <MenuBarBox>
      <MenuBarLogo>
        <Link to={'/'}><Logo /></Link>
      </MenuBarLogo>
      <MenuBarNav>
        {children}
      </MenuBarNav>
    </MenuBarBox>
  );
};

MenuBar.propTypes = {
  children: PropTypes.node,
  logo: PropTypes.any,
};

export {MenuBar};
