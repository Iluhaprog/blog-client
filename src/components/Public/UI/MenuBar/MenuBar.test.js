import {Apple} from '../Icons/Apple';
import {render} from '@testing-library/react';
import {MenuBar} from './MenuBar';
import React from 'react';
import {BrowserRouter} from 'react-router-dom';

test('Should render MenuBar', () => {
  const {asFragment} = render(
    <BrowserRouter>
      <MenuBar
        logo={Apple}
      >
        <a href={'#'}>Item1</a>
        <a href={'#'}>Item1</a>
        <a href={'#'}>Item1</a>
      </MenuBar>
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
