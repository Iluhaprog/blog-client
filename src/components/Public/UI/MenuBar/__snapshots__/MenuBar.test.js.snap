// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MenuBar 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTCUz icOZRy"
  >
    <div
      class="sc-dlfnbm IvnZy"
    >
      <a
        href="/"
      >
        <div
          class="sc-bdfBwQ jzjInw"
        >
          <svg>
            apple.svg
          </svg>
        </div>
      </a>
    </div>
    <nav
      class="sc-hKgILt hjUXII"
    >
      <a
        href="#"
      >
        Item1
      </a>
      <a
        href="#"
      >
        Item1
      </a>
      <a
        href="#"
      >
        Item1
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
